src/OrchardCore.Modules/OrchardCore.Workflows/Recipes/WorkflowTypeStep.cs

```diff
@@ -44,25 +46,32 @@ public async Task ExecuteAsync(RecipeExecutionContext context)
             }
 
             var model = context.Step.ToObject<WorkflowStepModel>();
+            var urlHelper = GetUrlHelper();
 
             foreach (var token in model.Data.Cast<JsonObject>())
             {
                 var workflow = token.ToObject<WorkflowType>(_jsonSerializerOptions);
 
-                foreach (var activity in workflow.Activities.Where(a => a.Name == nameof(HttpRequestEvent)))
-                {
-                    var tokenLifeSpan = activity.Properties["TokenLifeSpan"];
-                    if (tokenLifeSpan != null)
-                    {
-                        activity.Properties["Url"] = ReGenerateHttpRequestEventUrl(workflow, activity, tokenLifeSpan.ToObject<int>());
-                    }
-                }
-
                 var existing = await _workflowTypeStore.GetAsync(workflow.WorkflowTypeId);
 
-                if (existing == null)
+                if (existing is null)
```

Comment:
Why is pattern matching better here and below for `urlHelper` and `tokenLifeSpan`? These are single-operand expressions that are only more verbose than using `==` or `!=`.

Reply:
just slightly more readable.

Reply:
Maybe if you're a Basic developer, but here we're awesome C# people who can read `!=` just as well as `is not` :).

Reply:
yeah. search OC source and you'll see how many time `is` and `is not` is used :)

Reply:
![image](https://github.com/OrchardCMS/OrchardCore/assets/1976647/3523c5ff-c68d-43e1-ae8d-130be46be523)

But seriously: I get, enjoy, and use pattern matching where it makes sense, which is when there's actually a pattern to match against multiple operands vs only a binary expression as it is used here. Why would we use them in such cases? While technically equivalent, do you really think it's better to write `is` over `==` or `is not` over `!=`? The latter have been used for generations and every C# developer is entirely clear on them (not to mention the added issue that you can't use pattern matching for everything, so you'd have equality checks in both forms).

Reply:
when you read the code, it is shorted to say "without thinking" `is` vs `equal equal`. I am fine with `==` or `is` but the fact we already use it in many places, I thought why not.

Reply:
Pattern matching is quite useful, it reminds me of my baby VB.NET :)

Reply:
I'm all for pattern matching but in these cases none of its features were used, they were just literal (in)equality checks with a different syntax.

Since the other one is removed and I'm going on a vacation now, I'll leave the Basic developers to it :).